evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -242,6 +241,20 @@ def _transform_parameters(self, pipeline, proposed_parameters):
                         )
                     else:
                         component_parameters[param_name] = value
+            if name in self._pipeline_params and self._batch_number == 0:
+                for param_name, value in self._pipeline_params[name].items():
+                    if isinstance(value, (Integer, Real, Categorical)):
+                        raise ValueError(
+                            "Pipeline parameters should not contain skopt.Space variables, please pass them "
+                            "to custom_hyperparameters instead!"
+                        )
```

Comment:
This error is helpful!

I see `self._pipeline_params` here comes directly from `AutoMLSearch`, and is simply the `pipeline_params` input with the addition of any pipeline-level parameters (like time series config). I also see the `if` statements here only check the values of `self._pipeline_params`, and not the parameters proposed by the tuner.

So, we have a few options on where to do this check:
1. Don't do it
2. In `IterativeAlgorithm._transform_parameters` (current PR state)
3. In `IterativeAlgorithm.__init__`
4. In `AutoMLSearch.search`
5. In `AutoMLSearch.__init__`

Which do you think is best?

I think we should do this check, so 1 is out. I also think 3/4/5 are better than 2. Having this check in any of those spots is functionally equivalent -- if someone provides an skopt.Space in the `AutoMLSearch` `pipeline_parameters` input, this error will be triggered regardless. But in terms of code organization, it makes more sense to me to do this kind of check earlier. Since the check doesn't have anything to do with the proposed parameters here in `_transform_parameters`, I think its best to move it.

I think option 3 is my favorite. Doing it higher up the stack than `IterativeAlgorithm.__init__` means that initializing `IterativeAlgorithm` directly will skip the validation.